apache/incubator-metron
# METRON-339: Create YARN app to deploy endpoints

This is a Yarn application and associated utilities which is intended to allow users the ability to deploy REST endpoints (presumably to serve up data science models).

In order to test this, I would recommend the following:
* Deploy Metron on vagrant via the `quickdev`
* From monit @ `http://node1:2812` stop the following to ensure enough headroom (this is overkill, btw)
  * `yaf`
  * `snort`
  * `pcap-service`
  * `pcap-replay`
  * All of the parser topologies
  * The enrichment topology
  * The indexing topology
* ssh into node1 as root
* From node1
  * Create a directory `~/dummy`
  * Copy the [following](https://raw.githubusercontent.com/cestella/incubator-metron/METRON-339-1/metron-analytics/metron-maas-service/src/test/resources/maas/dummy_rest.sh) into the directory `~/dummy` as `dummy.sh`
  * Change directories to `/usr/metron/0.2.0BETA`
  * Start the `MaaS` service via `bin/maas_service.sh -zq node1:2181`
  * Start one instance of the dummy echo service (named `dummy` version `1.0` in a container of 500m via `bin/maas_deploy.sh -lmp ~/dummy -hmp /user/root/maas/dummy -m 500 -mo ADD -n dummy -ni 1 -v 1.0 -zq node1:2181`
  * Wait a couple seconds and ensure that it started by running `curl -i http://localhost:1500/echo/foobar` and ensuring you get `foobar` in return
  * List the active models via `bin/maas_deploy.sh -mo LIST -n dummy -zq node1:2181` and ensure that you see the `dummy` model show up in the output
  * Remove one instance of the dummy model via `bin/maas_deploy.sh -mo REMOVE -n dummy -ni 1 -v 1.0 -zq node1:2181`
  * After a couple seconds ensure that you cannot access the dummy model any longer via `curl -i http://localhost:1500/echo/foobar`
